test/e2e/tests/test_notebook_instance.py

```diff
@@ -146,7 +150,65 @@ def testCreateAndDelete(self, notebook_instance):
         )
         assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "True")
 
+    def update_notebook_test(self, notebook_instance):
+        (reference, resource, spec) = notebook_instance
```

Comment:
This test is now creating multiple instances of the notebook, have you verified that - 
1. All 3 of them are getting properly cleaned up ?
2. Is this increasing the test run time by a lot ?

You should be able to use the same notebook instance to run create, update and delete. Just change the scope of the pytest fixture `notebook_instance` to module and check if that is enough.

Reply:
Don't think it was creating 3 notebooks before but I still added the module scope just to be safe. The Notebook is properly cleaned up here https://github.com/aws-controllers-k8s/sagemaker-controller/blob/0d63e02e1aa5b32ec62811bf679d6e6eea429409/test/e2e/tests/test_notebook_instance.py#L54.